### DISCUSSION TITLE
Limit number of viewers of a stream to 1

### DISCUSSION THREAD

**Discussion by martiankoder** [OvenMediaLabs/OvenMediaEngine#1189] [Help]
Hi everyone,
Is there anything that can be set in the Server.xml file so that a stream can only be viewed by 1 viewer? If a second viewer  get the URL of the stream will not run. 
Also, is there an API to get the number of viewers watching a stream? Thanks

**Comment by gregfr** [comment]
Hello

If your viewers are accessing your streams with HLS, it can be quite difficult to know/limit the number of viewers, because HLS is an unconnected protocol based on HTTP (as opposed to RTMP which is connection-based).
In this case, to have a better control on who's accessing your streams, you better put OME behind a reverse proxy and have access control there.

Regarding OME itself, you can find the REST API there: https://airensoft.gitbook.io/ovenmediaengine/rest-api
Specifically, you can find some statistics here: https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/statistics/current

Regards

**Comment by bchah** [comment]
I would implement this in your application so you can control the logic yourself, perhaps leveraging AdmissionWebhooks to validate the individual playback requests against a one time “burn and turn” public key…

**Comment by getroot** [comment]
As @bchah mentioned you can do what you want with AdmissionWebhooks.
https://airensoft.gitbook.io/ovenmediaengine/access-control/admission-webhooks
